colorls.rb

```diff
@@ -124,21 +172,28 @@ def options(content)
   end
 end
 
-args = *ARGV
+args         = *ARGV
+report       = false
+sort         = false
+one_per_line = false
 
-if args.include?('--report') || args.include?('-r')
-  report = true
-  args -= %w[--report -r]
-else
-  report = false
+args.each do |arg|
+  report       = true if ['--report', '-r'].include?(arg)
+  one_per_line = true if arg == '-1'
+
+  match = arg.match(/^--sort=?(.*)?$/)
+
+  if match
+    sort = match.captures[0] == 'dirs-first' ? 'dirs-first' : true
+  end
 end
 
 args.keep_if { |arg| !arg.start_with?('-') }
 
 if args.empty?
-  ColorLS.new(nil, report).ls
+  ColorLS.new(nil, report, sort, one_per_line).ls
 else
-  args.each { |path| ColorLS.new(path, report).ls }
+  args.each { |path| ColorLS.new(path, report, sort, one_per_line).ls }
 end
```

Comment:
Now that number of variables are increasing, order of variables might turn out to be a pain later. Add keyword args please. Like, `ColorLS.new(path, report: report, sort: sort, per_line: 1 or false).ls`